### ISSUE TITLE
Rule update API (PUT) is not idempotent anymore because of PR #54515

### ISSUE THREAD

**Issue by Cakasim** [getsentry/sentry#55438]
### Environment

SaaS (https://sentry.io/)

Might be related to: #54515 

### Steps to Reproduce

We are using the Terraform provider to configure Sentry. The Terraform apply stopped working because of the mentioned API change. The provider attempts to PUT the existing rule, according to REST principles this should be fine since PUT should behave idempotently.

Potential steps to recreate (without Terraform):
1. Use the API to create any rule
2. Send a PUT request with unchanged data
3. API should return 400 because of duplicate alert

### Expected Result

1. Use the API to create a rule
2. Send a PUT request with unchanged rule data
3. Get 2xx code because no changes
4. Requesting multiple times should not result in an error

### Actual Result

```
Error: PUT https://sentry.io/api/0/projects/xxxxx/xxxxxx/rules/12xxxxxx/: 400 map[name:[This rule is an exact duplicate of 'staging - errors' in this project and may not be created.]]
│ 
│   with sentry_issue_alert.errors["staging"],
│   on main.tf line 76, in resource "sentry_issue_alert" "errors":
│   76: resource "sentry_issue_alert" "errors" {
│ 
```

### Product Area

APIs

**Comment by Dhrumil-Sentry** [comment]
@ceorourke @scefali could you please take a look

**Comment by ceorourke** [comment]
Hi @Cakasim I think this was already fixed in https://github.com/getsentry/sentry/pull/55361 

You're talking about (if it's happening in the app) creating a rule, hitting edit, making no changes, and hitting save again, right? That works w/o raising an error about duplication.

**Comment by Cakasim** [comment]
> Hi @Cakasim I think this was already fixed in #55361
> 
> You're talking about (if it's happening in the app) creating a rule, hitting edit, making no changes, and hitting save again, right? That works w/o raising an error about duplication.

Hi @ceorourke, yeah seems like it is fixed. The error disappeared for us. Thank you for clarification!
